packages/container/lib/registry.js

```diff
@@ -691,6 +692,32 @@ Registry.prototype = {
     });
   },
 
+  /**
+   @method eachForType
+   @param {String} type the type to iterate over
+   @param {Function} callback the function to invoke for each item
+   @private
+  */
+  eachForType(type, callback) {
+    let registeredNames = keys(this.registrations);
+    for (let index = 0, length = registeredNames.length; index < length; index++) {
+      let fullName = registeredNames[index];
+      let itemType = fullName.split(':')[0];
+
+      if (itemType === type) {
+        callback(fullName, this.registrations[fullName]);
+      }
+    }
+
+    if (this.fallback) {
+      this.fallback.eachForType(type, callback);
+    }
+
+    if (this.resolver.eachForType) {
+      this.resolver.eachForType(type, callback);
```

Comment:
If you walk both the container and the resolver (or container and fallback), will you sometimes get duplicates? Things already loaded will show up twice?